src/D2L.CodeStyle.Analyzers/Async/Generator/SyncGenerator.cs

```diff
@@ -0,0 +1,182 @@
+	}
+
+	private static bool IsInterestingLookingMethod(
+		SyntaxNode node,
+		CancellationToken _
+	) {
+		if( node is not MethodDeclarationSyntax method ) {
+			return false;
+		}
+
+		// We only care about async methods
+		if( !method.Modifiers.Any( t => t.IsKind( SyntaxKind.AsyncKeyword ) ) ) {
+			return false;
+		}
+
+		// The ones we care about have to have a particular attribute -- so
+		// perk up when we see _any_ attributes (we don't have a SemanticModel
+		// in this method.)
+		return method.AttributeLists.Any();
+	}
+
+	private static MethodDeclarationSyntax? ExtractSyntax(
+		GeneratorSyntaxContext context,
+		CancellationToken cancellationToken
+	) {
+		var methodDeclaration = (MethodDeclarationSyntax)context.Node;
+
+		var attributes = methodDeclaration.AttributeLists
+			.SelectMany( al => al.Attributes );
+
+		foreach( var attribute in attributes ) {
+			// We have the semantic model now, so make sure this method has
+			// the right attribute.
+			if( IsGenerateSyncAttribute( context.SemanticModel, attribute, cancellationToken ) ) {
+				return methodDeclaration;
+			}
+		}
+
+		return null;
+	}
+
+	internal static bool IsGenerateSyncAttribute(
+		SemanticModel model,
+		AttributeSyntax attribute,
+		CancellationToken cancellationToken
+	) {
+		var attributeConstructorSymbol = model.GetSymbolInfo( attribute, cancellationToken ).Symbol as IMethodSymbol;
+
+		if( attributeConstructorSymbol == null ) {
+			return false;
+		}
+
+		return attributeConstructorSymbol.ContainingType.ToDisplayString()
+			== "D2L.CodeStyle.Annotations.GenerateSyncAttribute";
+	}
+
+	private static MethodGenerationResult? GenerateSyncMethod(
+		(MethodDeclarationSyntax MethodDeclaration, Compilation Compilation) data,
+		CancellationToken cancellationToken
+	) {
+		if( !data.Compilation.ContainsSyntaxTree( data.MethodDeclaration.SyntaxTree ) ) {
+			return null;
+		}
+
+		var model = data.Compilation.GetSemanticModel( data.MethodDeclaration.SyntaxTree );
+		var methodSymbol = model.GetDeclaredSymbol( data.MethodDeclaration, cancellationToken );
+
+		if( methodSymbol == null || methodSymbol.Kind == SymbolKind.ErrorType ) {
+			return null;
+		}
+
+		var transformer = new AsyncToSyncMethodTransformer( model, cancellationToken );
+
+		var transformResult = transformer.Transform( data.MethodDeclaration );
+
+		return new(
+			Original: data.MethodDeclaration,
+			GeneratedSyntax: transformResult.Success ? transformResult.Value!.ToFullString() : "",
+			Diagnostics: transformResult.Diagnostics
+		);
+	}
+
+	private static IEnumerable<FileGenerationResult> GenerateFiles(
+		ImmutableArray<MethodGenerationResult> methodResults,
+		CancellationToken cancellationToken
+	) {
+		var methodsByFile = methodResults
+			.GroupBy( static mr => mr.Original.SyntaxTree.GetCompilationUnitRoot() )
+			.OrderBy( static mr => mr.First().Original.SyntaxTree.FilePath )
+			.ToImmutableArray();
+
+		var hintNames = SourceFilePathHintNameGenerator.Create(

```

Comment:
nit: Had to look it up when reading. Also needs a using if accepted.

```suggestion
		IEnumerable<string> hintNames = SourceFilePathHintNameGenerator.Create(
```